### PR TITLE
docs: describe dev-env setup for Windows machines

### PR DESCRIPTION
**Pull request by JanMalch**
Windows machines do not have the required `patch` command available.
The added step explains how to install the package on Windows.

Refs #17456

### CODE REVIEW

**Comment by josephperrott** [comment]
Assigning @devversion as the windows expert.

File: DEV_ENVIRONMENT.md [hunk 1]
```diff
@@ -2,12 +2,16 @@
 
 1. Make sure you have `node` installed with a version at _least_ 10.0.0 and `yarn` with a version
    of at least 1.10.0. We recommend using `nvm` to manage your node versions.
-2. Run `yarn global add gulp` to install gulp.
-3. Fork the `angular/components` repo on GitHub.
-4. Clone your fork to your machine with `git clone`.
+2. Make sure the `patch` command is available
+   - On Windows: Install [`MSYS2`](https://www.msys2.org/) by following the described setup.
+     Afterwards you can install the `patch` package by running `pacman -S patch`.
+     Add `C:\msys64\usr\bin` to the `PATH` environment variable.
```
Comment (devversion) [inline DEV_ENVIRONMENT.md]: Can we somehow link to this guide for the Bazel installation? https://docs.bazel.build/versions/master/install-windows.html#5-optional-install-compilers-and-language-runtimes

Comment (JanMalch) [reply DEV_ENVIRONMENT.md]: I rephrased it so it's more fitting for the actual problem

File: DEV_ENVIRONMENT.md [hunk 2]
```diff
@@ -2,12 +2,16 @@
 
 1. Make sure you have `node` installed with a version at _least_ 10.0.0 and `yarn` with a version
    of at least 1.10.0. We recommend using `nvm` to manage your node versions.
-2. Run `yarn global add gulp` to install gulp.
-3. Fork the `angular/components` repo on GitHub.
-4. Clone your fork to your machine with `git clone`.
+2. Material uses Bazel which requires certain Bash and UNIX tools.
```
Comment (jelbourn) [inline DEV_ENVIRONMENT.md]:
```suggestion
2. angular/components uses Bazel which requires certain Bash and UNIX tools.
```

Comment (JanMalch) [reply DEV_ENVIRONMENT.md]: Should it be changed further down aswell?

Comment (jelbourn) [reply DEV_ENVIRONMENT.md]: Ah yes, I missed then when renaming the repo

**Pull request merged**